pytest/tests/mocknet/run_adversenet.py

```diff
@@ -83,13 +83,39 @@ def check_slow_blocks(initial_metrics, final_metrics):
 
 def override_config(node, config):
     # Add config here depending on the specific node build.
-    pass
-    """
-    if "bad" in node.instance_name:
-        config["adversarial"] = {
-            "produce_duplicate_blocks": True
+
+    # Enable the following if you want to increase the block time
+    # from the default 0.6s to 1s.
+    if False:
+        config["consensus"]["min_block_production_delay"] = {
+            "secs": 1,
+            "nanos": 0,
+        }
+
+    # Enable the following if testing single-shard tracking.
+    if False:
+        config["state_sync"] = {
+            "dump": {
+                "location": {
+                    "GCS": {
+                        "bucket": "near-mocknet-snet2-state"
+                    }
+                },
+                "iteration_delay": {
+                    "secs": 1,
+                    "nanos": 0
+                },
+            }
+        }
```

Comment:
This is the config for the state dumper node. In your setup you will need to have at least one of them. The node need to track all shards to be able to generate the headers.
```suggestion
        config["state_sync"] = {
          "sync": {
            "ExternalStorage": {
              "location": {
                "GCS": {
                  "bucket": "near-mocknet-snet2-state"
                }
              },
              "num_concurrent_requests": 25,
              "num_concurrent_requests_during_catchup": 5
            }
          }
        }
```

Reply:
Huh, I wonder how did this work at all then; the nodes did do state sync 🤔